src/CabanaPD_Particles.hpp

```diff
@@ -443,25 +430,29 @@ class Particles<MemorySpace, PMB, TemperatureIndependent, Dimension>
             return sliceCurrentPosition();
     }
 
+    template <typename... OtherFields>
     void output( [[maybe_unused]] const int output_step,
                  [[maybe_unused]] const double output_time,
-                 [[maybe_unused]] const bool use_reference = true )
+                 [[maybe_unused]] const bool use_reference,
+                 [[maybe_unused]] OtherFields&&... other )
     {
         _output_timer.start();
 
 #ifdef Cabana_ENABLE_HDF5
         Cabana::Experimental::HDF5ParticleOutput::writeTimeStep(
             h5_config, "particles", MPI_COMM_WORLD, output_step, output_time,
-            n_local, getPosition( use_reference ), sliceStrainEnergy(),
-            sliceForce(), sliceDisplacement(), sliceVelocity(), sliceDamage() );
+            n_local, getPosition( use_reference ), sliceForce(),
```

Comment:
You have a variadic output() function that defaults to writing out force, displacement, and velocity, and any others provided for a Particles that is tagged as `NoOutput`. I would rename the tag as `BaseOutput` or something else.